src/redux/books/books.js

```diff
@@ -0,0 +1,124 @@
+/* eslint-disable camelcase */
+import axios from 'axios';
+
+const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/JRvgmQWxBSuTgawpguvn/books/';
+const REMOVE_BOOK_SUCCESS = 'REMOVE_BOOK_SUCCESS';
+const ADD_BOOK_SUCCESS = 'ADD_BOOK_SUCCESS';
+const GET_BOOK_REQUEST = 'GET_BOOK_REQUEST';
+const GET_BOOK_SUCCESS = 'GET_BOOK_SUCCESS';
+const GET_BOOK_FAILURE = 'GET_BOOK_FAILURE';
+
+export const getBookRequest = () => ({
+  type: GET_BOOK_REQUEST,
+});
+
+export const getBookSuccess = (books) => ({
+  type: GET_BOOK_SUCCESS,
+  payload: books,
+});
+
+export const getBookFailure = (error) => ({
+  type: GET_BOOK_FAILURE,
+  payload: error,
+});
+
+export const addBookSuccess = (payload) => ({
+  type: ADD_BOOK_SUCCESS,
+  payload,
+});
+
+export const removeBookSuccess = (payload) => ({
+  type: REMOVE_BOOK_SUCCESS,
+  payload,
+});
+
+export const fetchBooks = () => (dispatch) => {
+  dispatch(getBookRequest);
+  axios.get(url)
+    .then((response) => {
+      const books = Object.entries(response.data).map((book) => ({
+        id: book[0],
+        title: book[1][0].title,
+        category: book[1][0].category,
+        author: 'John Doe',
+      }));
+      dispatch(getBookSuccess(books));
+    })
+    .catch((error) => {
+      const errorMsg = error.message;
+      dispatch(getBookFailure(errorMsg));
+    });
+};
+
+export const addBook = (newBook) => (dispatch) => {
+  const { item_id, category, title } = newBook;
+  axios.post(url, {
+    item_id,
+    title,
+    category,
+  }).then(() => {
+    dispatch(addBookSuccess({
+      id: item_id,
+      title,
+      category,
+      author: 'Jane Doe',
+    }));
+  });
+};
+
+export const removeBook = (id) => (dispatch) => {
+  fetch(`url${id.id}`, {
+    method: 'DELETE',
+    body: JSON.stringify({
+      item_id: id,
+    }),
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  })
+    .then(() => {
+      dispatch(removeBookSuccess(id));
+    });
+};
+
```

Comment:
- [ ] The `Add book` and `remove book` features should work the same as before even after the refactoring to use the API has been done currently, I am trying to add a book but it can't post please fix that

Reply:
This functionality has been implemented well